### ISSUE TITLE
Opening unversioned file shows git error

### ISSUE THREAD

**Issue by dnicolson** [timbrel/GitSavvy#1458]
To reproduce open any unversioned file via the file menu or command line:

```
cd ~
touch file
subl file
```

A git repository does not exist so the following error is displayed:
```
$ git status --porcelain -z -b

fatal: not a git repository (or any of the parent directories): .git
```

<img width="909" alt="Screen Shot 2021-03-24 at 18 54 57" src="https://user-images.githubusercontent.com/2276355/112360719-e486ea80-8cd2-11eb-9da4-cd3bdebc0eb1.png">

**Comment by dnicolson** [comment]
The issue appears to have been introduced in https://github.com/timbrel/GitSavvy/commit/69d36a8.

**Comment by kaste** [comment]
Hm, unfortunate.  

What we do here is search the path upwards for a `.git` file or folder.  It looks like we find one, maybe at `~/.git`.  Can you enable the logging via Command Palette `GitSavvy: enable logging`, then open the Sublime console.  After that open an untracked file, it should log "Searching .git upwards, starting at ..." and so on.

**Comment by dnicolson** [comment]
There is a `~/.git` directory, I'm not seeing any logging in the console though.

**Comment by kaste** [comment]
Yeah, okay, we don't validate the ".git"-directories we find.  What's the reason for you having it?  Should we validate?

**Comment by dnicolson** [comment]
Perhaps, `~/.git` is not a repo but seems to be used for configuration. [Watchman](https://github.com/facebook/watchman) and [Gitfox](https://www.gitfox.app/) appear to do this:

```
tree -a .git
.git
├── .watchman-cookie-Daves-MacBook-Pro.local-852-63
├── .watchman-cookie-Daves-MacBook-Pro.local-852-64
├── .watchman-cookie-Daves-MacBook-Pro.local-852-65
├── .watchman-cookie-Daves-MacBook-Pro.local-852-66
├── .watchman-cookie-Daves-MacBook-Pro.local-852-67
├── .watchman-cookie-Daves-MacBook-Pro.local-852-68
└── gitfox
    └── defaults.plist
```

**Comment by kaste** [comment]
According to the git documenation a *.git* dir should have a "HEAD" file.  So we validate now by looking HEAD.   git does something similar https://github.com/git/git/blob/f9f2520108bab26a750bcbb00518dc27672cf0a2/setup.c#L339-L339 

Fix ships with v2.29.1
